conftest.py

```diff
@@ -455,6 +458,57 @@ def service_actions():
         httpd.serve_forever()
 
 
+def run_in_pyodide(*args):
+    # This can be called as @run_in_pyodide or @run_in_pyodide("first_dep",
+    # "second_dep"). In the former case, run_in_pyodide acts as the decorator.
+    # In the latter, it calls run_in_pyodide("first_dep", "second_dep") and
+    # treats the return value as a decorator.
+
+    # direct_call is true if it is the first case
+    direct_call = len(args) == 1 and callable(args[0])
+
+    def decorator(f):
+        def wrapped(selenium):
+            if not direct_call:
+                selenium.load_package(list(args))
+            lines, start_line = inspect.getsourcelines(f)
+            # Remove first line, which is the decorator. Then pad with empty lines to fix line number.
+            lines = ["\n"] * start_line + lines[1:]
+            source = "".join(lines) + f"\n{f.__name__}()"
+
+            msg = None
+            try:
+                selenium.run(source)
+            except WebDriverException as e:
+                msg = str(e)
+                if msg.startswith("Message: Error: Traceback (most recent call last)"):
+                    # This is a python traceback
+                    # Remove first 4 lines of filler
+                    msg = msg.splitlines()[4:]
+
+                    # Replace <exec> with file name of actual source
+                    filename = inspect.getsourcefile(f)
+                    msg = [x.replace('File "<exec>"', f'File "{filename}"', 1) for x in msg]
+
+                    # Join them back
+                    msg = "Traceback (most recent call last):\n" + "\n".join(msg)
+
+            # Print outside for better stack trace
+            if msg is not None:
+                pytest.fail(
+                    "Error running function in pyodide\n\n" + msg,
+                    pytrace=False,
+                )
+
+        return wrapped
+
+    if direct_call:
+        return decorator(args[0])
+    else:
+        return decorator
+
+builtins.run_in_pyodide = run_in_pyodide
```

Comment:
Assuming there is no closure on the selenium fixture object, we could just put this in some normal python module. Maybe `pyodide_build/testing.py` that way it's a bit less magical?

Reply:
That would work too